widget/playground/src/containers/DefaultChainAndToken/DefaultChainAndToken.tsx

```diff
@@ -0,0 +1,124 @@
+import type { Type } from '../../types';
+
+import { ChainsIcon, Divider } from '@rango-dev/ui';
+import React, { useState } from 'react';
+
+import { ItemPicker } from '../../components/ItemPicker';
+import { OverlayPanel } from '../../components/OverlayPanel';
+import { SingleList } from '../../components/SingleList';
+import { tokensAreEqual, tokenToString } from '../../helpers';
+import { useConfigStore } from '../../store/config';
+import { useMetaStore } from '../../store/meta';
+import { ModalState } from '../FunctionalLayout/FunctionalLayout.types';
+
+export function DefaultChainAndToken({ type }: { type: Type }) {
+  const [modalState, setModalState] = useState<ModalState | null>(null);
+  const {
+    config: { from, to },
+    onChangeBlockChain,
+    onChangeToken,
+  } = useConfigStore();
+  const {
+    meta: { blockchains, tokens },
+  } = useMetaStore();
+
+  const selectedType = type === 'Source' ? from : to;
+
+  const filteredTokens = selectedType?.blockchain
+    ? tokens.filter((token) => token.blockchain === selectedType.blockchain)
+    : [];
+  const chainValue = blockchains.find(
+    (chain) => chain.name === selectedType?.blockchain
+  );
+  const tokenValue = tokens.find((token) =>
+    tokensAreEqual(token, selectedType?.token)
+  );
+
+  const handleDefaultBlockchainConfirm = (item: string) => {
+    onChangeBlockChain(item, type);
+    onChangeToken(undefined, type);
+
+    onBack();
+  };
+
+  const handleDefaultTokenConfirm = (item: string) => {
+    const selectedToken = filteredTokens.find(
+      (token) => tokenToString(token) === item
+    );
+    if (selectedToken) {
+      onChangeToken(
+        {
+          blockchain: selectedToken.blockchain,
+          address: selectedToken.address,
+          symbol: selectedToken.symbol,
+        },
+        type
+      );
+    }
+
+    onBack();
+  };
+
+  const onBack = () => setModalState(null);
+
+  const filteredBlockchains = selectedType?.blockchains
+    ? blockchains.filter((chain) =>
+        selectedType.blockchains?.includes(chain.name)
+      )
+    : blockchains;
+
+  return (
+    <>
+      <ItemPicker
+        onClick={() => setModalState(ModalState.DEFAULT_BLOCKCHAIN)}
+        value={{ label: chainValue?.displayName, logo: chainValue?.logo }}
+        title="Default Blockchain"
+        hasLogo={true}
+        placeholder="Chain"
+      />
+      <Divider size={12} />
+      <ItemPicker
+        onClick={() => setModalState(ModalState.DEFAULT_TOKEN)}
+        value={{ label: tokenValue?.symbol, logo: tokenValue?.image }}
+        title="Default Token"
+        hasLogo={true}
+        placeholder="Token"
+        disabled={!chainValue}
+      />
+      {modalState === ModalState.DEFAULT_BLOCKCHAIN && (
+        <OverlayPanel onBack={onBack}>
+          <SingleList
```

Comment:
When I want to change the blockchain for the second time, we have some bugs:

![Recording 2023-10-21 at 12 29 55](https://github.com/rango-exchange/rango-client/assets/99180934/c530a209-af23-4647-b648-cf16c96ba979)

Reply:
I fixed it but it should be refactored in embedded.